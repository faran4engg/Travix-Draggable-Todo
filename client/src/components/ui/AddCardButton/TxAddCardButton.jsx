import React, { useState } from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import TxForm from "../Form/TxForm";

const Anchor = styled.a`
  cursor: pointer;
  opacity: 1;
  color: white;
  &:hover {
    opacity: 1;
    color: #fff;
  }
  span:hover {
    text-decoration: underline;
    color: #fff;
  }
`;

const renderForm = (listName, setIsFormOpen) => (
  <TxForm onCancel={() => setIsFormOpen(false)} listName={listName} />
);

const renderAddButton = setIsFormOpen => (
  <Anchor className="item" onClick={() => setIsFormOpen(true)}>
    <i aria-hidden="true" className="add square icon"></i>
    <span>Add a Card</span>
  </Anchor>
);

const TxAddCardButton = props => {
  const { listName } = props;
  const [isFormOpen, setIsFormOpen] = useState(false);
  
  return isFormOpen
    ? renderForm(listName, setIsFormOpen)
    : renderAddButton(setIsFormOpen);
};

TxAddCardButton.propTypes = {
  listName: PropTypes.string
};

export default TxAddCardButton;
