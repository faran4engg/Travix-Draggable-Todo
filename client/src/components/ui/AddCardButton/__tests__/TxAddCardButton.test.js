import React from "react";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import { mount } from "enzyme";
import TxAddCardButton from "../TxAddCardButton";

describe("TxAddCardButton", () => {
  it("should render correctly ", () => {
    const component = mount(<TxAddCardButton listName="todo" />);
    expect(component.find("a")).toHaveLength(1);
    expect(component).toMatchSnapshot();
  });
  
  describe("when clicked on Add Card button", () => {
    it("it should display the form", () => {
      const store = configureStore()();
      const component = mount(
        <Provider store={store}>
          <TxAddCardButton listName="todo" />
        </Provider>
      );
      const anchor = component.find("a");
      anchor.simulate("click");
      expect(component.find("TxForm")).toHaveLength(1);
    });
  });
});
