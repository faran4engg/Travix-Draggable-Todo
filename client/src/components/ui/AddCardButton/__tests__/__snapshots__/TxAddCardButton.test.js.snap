// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TxAddCardButton should render correctly  1`] = `ReactWrapper {}`;

exports[`TxInput should render correctly  1`] = `ShallowWrapper {}`;
